es/es-server/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -189,41 +173,37 @@ void start() {
                 final DiscoveryNode node = nodes[i];
                 final String nodeId = node.getId();
                 try {
-                    if (node == null) {
```

Comment:
This `if (node == null)` case didn't make sense, there would already have been a NPE in 174 where the `nodeId` is retrieved from `node`.